src/eos-updater-poll-lan.c

```diff
@@ -0,0 +1,920 @@
+      const gchar *sha512sum_local;
+
+      sha512sum = g_hash_table_lookup (records, eos_avahi_v1_branch_file_sha512sum ());
+      sha512sum_local = lan_data->fetch_data->data->branch_file->contents_sha512sum;
+      if (g_strcmp0 (sha512sum_local, sha512sum) == 0)
+        eos_service_with_branch_file_get_from_lan_data (swbf, lan_data);
+      else
+        swbf->declared_sha512sum = g_strdup (sha512sum);
+      result = SERVICE_VALID;
+    }
+  if (result == SERVICE_VALID)
+    return TRUE;
+  return FALSE;
+}
+
+static gboolean
+txt_v1_handler (LanData *lan_data,
+                EosServiceWithBranchFile *swbf,
+                gboolean *valid,
+                GError **error)
+{
+  g_autoptr(GHashTable) records = NULL;
+  TxtRecordError txt_error = get_unique_txt_records (swbf->service->txt,
+                                                     &records,
+                                                     eos_avahi_v1_ostree_path (),
+                                                     eos_avahi_v1_branch_file_dl_time (),
+                                                     eos_avahi_v1_branch_file_sha512sum (),
+                                                     NULL);
+  if (txt_error != TXT_RECORD_OK)
+    {
+      // TODO: message
+      *valid = FALSE;
+      return TRUE;
+    }
+  if (!txt_v1_handler_risky_checks (lan_data,
+                                    records,
+                                    valid,
+                                    error))
+    return FALSE;
+  if (!*valid)
+    return TRUE;
+  *valid = txt_v1_handler_checks (lan_data,
+                                  swbf,
+                                  records);
+  return TRUE;
+}
+
+static gboolean
+txt_v2_handler_risky_checks (LanData *lan_data,
+                             GHashTable *records,
+                             gboolean *valid,
+                             GError **error)
+{
+  if (!check_ostree_path (lan_data,
+                          g_hash_table_lookup (records, eos_avahi_v2_ostree_path ()),
+                          valid,
+                          error))
+    return FALSE;
+  if (!*valid)
+    // TODO: message
+    return TRUE;
+  return TRUE;
+}
+
+static gboolean
+txt_v2_handler_checks (LanData *lan_data,
+                       EosServiceWithBranchFile *swbf,
+                       GHashTable *records)
+{
+  const gchar *dl_time;
+  g_autoptr(GDateTime) txt_utc = NULL;
+
+  dl_time = g_hash_table_lookup (records, eos_avahi_v2_branch_file_timestamp ());
+  switch (check_dl_time (lan_data, dl_time, &txt_utc))
+    {
+    case DL_TIME_OLD:
+      eos_service_with_branch_file_get_from_lan_data (swbf, lan_data);
+      return TRUE;
+
+    case DL_TIME_NEW:
+      swbf->declared_download_time = g_steal_pointer (&txt_utc);
+      return TRUE;
+
+    case DL_TIME_INVALID:
+      // TODO: message
+      return FALSE;
+
+    case DL_TIME_OUT_OF_RANGE:
+      // TODO: message
+      return FALSE;
+    }
```

Comment:
I wrote it like that, because I thought I'll change the semantics of the second version. But in the end they have some similarities. Will split it out to a separate function.